Lannister/App/Controllers/DashboardController.swift

```diff
@@ -274,17 +287,30 @@ extension DashboardController : UICollectionViewDataSource {
             let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "holdingCellId", for: indexPath) as! HoldingCell
             
             let holding = holdings[indexPath.row-1]
-            let currency = holding.currency
             cell.colorView.backgroundColor = Colors.hexStringToUIColor(hex: holding.hexColor)
             cell.nameLabel.text = holding.name
-            let formattedNumber = numberFormatter.string(for: NSNumber(value: holding.value!))
-            let value = String(format: "%@", formattedNumber ?? "--")
-            cell.valueLabel.text = "\(currency!.symbol!)\(value)"
-            var holdingValue = holding.value
-            if holding.value < 0 {
+            var holdingValue : Double? = holding.value
+            var holdingCurrency = holding.currency
+            if holding.address != nil {
+                // Get ETH
+                let predicateAddress = NSPredicate(format: "address == %@", holding.address!)
```

Comment:
This code seems to a duplicate of line https://github.com/lannister-capital/lannister-ios/pull/54/files#diff-1363ef2460492ec1e1aed836a15f02e1R129. Can we refactor this?

Reply:
yes, and we should 👍